### PR TITLE
    Fix F841 - Unused variables

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:
None

### Summary of the issue:
NVDA's code contains unused variables, with specific directives for the linter to ignore them (`#noqa F841`)

### Description of user facing changes
There is no more `#noqa F841` directive in the code.

### Description of development approach
Searched for `F841` in the code and solved the situations:
1. Generally, removed the unused variable and checked that its presence without any use was not due to another oversight to use it elsewhere
2. 2. Or keep the previously unused variable and use it where it was aimed to (e.g. in `NVDAObjects/window/_msOfficeChart.py`)
3. In `winConsoleHandler.py`, renamed `consoleEventHookHandles` to `consoleWinEventHookHandles` since the goal clearly seems to empty the list.

Also used a Flake8 report on an older version of NVDA to identify other `F841` issues. This way, I have been able to find undefined private variables (i.e. starting with underscore) that are not caught by Ruff's default rule.

### Testing strategy:
Manual checks:
* Excel Charts
* Windows console? (TBD)

### Known issues with pull request:
In `WordDocumentTextInfo._normalizeFormatField`, the RUFF linter did not complain that `_startOffset` and `_endOffset` were not used, even if there is no `#noqa F841` directives on these lines. It's because Ruff's default config exclude private variables (i.e. starting with underscore) from unused variable checking.
I wonder why there is such default config for Ruff and if excluding private variables from the check may prevent us from finding potential unused variable and related issues.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [ ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved parsing within the Doxyfile processing to enhance reliability.
  - Refined description handling in MS Office objects for better accuracy.
  - Fixed candidate description formatting logic in behaviors.
  - Streamlined series name retrieval in MS Office Chart handling for performance.
  - Optimized selection offsets and embedded object labeling in window edits.
  - Simplified field normalization in Winword processing.
  - Enhanced handling of Outlook meeting/appointment creation forms.
  - Improved logic in PowerPoint controller for role assignments and overlap info.

- **Refactor**
  - Removed unused variable assignments across various modules to streamline code and improve performance.

- **Tests**
  - Simplified test code in `test_winVersion.py` by removing unused variable assignments for clarity and maintainability.

### CODE REVIEW

File: source/brailleDisplayDrivers/papenmeier.py [hunk 1]
```diff
@@ -152,7 +152,6 @@ def connectBluetooth(self):
 				key=lambda item: "bluetoothName" in item,
 			):
 				port = portInfo["port"]
-				hwID = portInfo["hardwareID"]  # noqa: F841
```
Comment (CyrilleB79) [inline source/brailleDisplayDrivers/papenmeier.py:155]: @jcsteh or @LeonarddeR, I do not know braille driver code at all.
Would you be able to confirm that this variable was not meant to be used elsewhere but forgotten instead?

File: source/winConsoleHandler.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ def disconnectConsole():
 	# Unregister any win events we are using
 	for handle in consoleWinEventHookHandles:
 		winUser.unhookWinEvent(handle)
-	consoleEventHookHandles = []  # noqa: F841
+	consoleWinEventHookHandles = []
```
Comment (CyrilleB79) [inline source/winConsoleHandler.py:112]: @michaelDCurran, @codeofdusk or @jcsteh:
Before this change, the list `consoleWinEventHookHandles` was not cleared. This means that we were potentially calling `winUser.unhookWinEvent` on handles of events already unhooked. Could this have caused any noticeable bug?

File: source/brailleDisplayDrivers/papenmeier.py (hunk 1 above)
Comment (LeonarddeR) [reply source/brailleDisplayDrivers/papenmeier.py:155]: Pretty sure this is safe to drop if not used elsewhere.

**Comment by LeonarddeR** [comment]
> `#noqa F841` only remains where necessary.

If there are any remaning variables, could you please underscore prefix them and remove the noqa?

> In `WordDocumentTextInfo._normalizeFormatField`, the RUFF linter did not complain that `_startOffset` and `_endOffset` were not used, i.e. there is no `#noqa F841` directives on these lines.

If this is an issue that can be isolated pretty easily, it might be worthwhile to report this to Ruff.

**Comment by CyrilleB79** [comment]
> > `#noqa F841` only remains where necessary.
> 
> If there are any remaning variables, could you please underscore prefix them and remove the noqa?

I have removed the last one in commit f2a7f741f966ca1495d2d78b60b3cebc9eb546a4. Even if it had been introduced on purpose (cc @josephsl), there was no need to define an unused variable in this unit test.


> > In `WordDocumentTextInfo._normalizeFormatField`, the RUFF linter did not complain that `_startOffset` and `_endOffset` were not used, i.e. there is no `#noqa F841` directives on these lines.
> 
> If this is an issue that can be isolated pretty easily, it might be worthwhile to report this to Ruff.

I am just realizing that Ruff ignores private variables(i.e. prefixed with underscore) by default, what Flake8 probably did not do. Why is there such default in Ruff? If we stick with this default configuration, we will miss unused private variables (prefixed with underscore), and thus miss any potential issue.

**Comment by LeonarddeR** [comment]
> I am just realizing that Ruff ignores private variables(i.e. prefixed with underscore) by default, what Flake8 probably did not do. Why is there such default in Ruff? If we stick with this default configuration, we will miss unused private variables (prefixed with underscore), and thus miss any potential issue.

I believe they only ignored them for F841. We could fix this by overriding [dummy-variable-rgx](https://docs.astral.sh/ruff/settings/#lint_dummy-variable-rgx)

**Comment by CyrilleB79** [comment]
> > I am just realizing that Ruff ignores private variables(i.e. prefixed with underscore) by default, what Flake8 probably did not do. Why is there such default in Ruff? If we stick with this default configuration, we will miss unused private variables (prefixed with underscore), and thus miss any potential issue.
> 
> I believe they only ignored them for F841.

Yes, that's what I was meaning.

> We could fix this by overriding [dummy-variable-rgx](https://docs.astral.sh/ruff/settings/#lint_dummy-variable-rgx)

Of course. But before overriding the default rule, in the first place, I would be curious to know the reason why they ignore private variables.

**Comment by LeonarddeR** [comment]
I guess this is by design/convension. For example if you want to loop ten times without needing the iteration variable, you can do `for _i in range(10)` without the linter complaining.

**Comment by josephsl** [comment]
Hi,

Ah, the nonexistent Windows version variable...

I did it to improve readability. If assigning a dedicated variable impedes readablity, then I guess we should remove it but with justifications.

Thanks.

**Comment by CyrilleB79** [comment]
> Ah, the nonexistent Windows version variable...
> 
> I did it to improve readability. If assigning a dedicated variable impedes readablity, then I guess we should remove it but with justifications.

I have finally removed the unused variable (see commit f2a7f741f966ca1495d2d78b60b3cebc9eb546a4).
To me, the code already seems extra clear and self explaining as is:
* the name `test_getWinVerFromNonExistentRelease` describes well what the test does
* the comment indicate more details to explain more
* and the member of the enum (`winVersion.WIN10_2003`) is also very self explaining

@josephsl, which extra explanation do you expect? Something more in the test's code (other comment, etc.)? Or another explanation comment on this page? Please be more specific if needed. Thanks.

Or 

The code for this

**Comment by CyrilleB79** [comment]
> > I am just realizing that Ruff ignores private variables(i.e. prefixed with underscore) by default, what Flake8 probably did not do. Why is there such default in Ruff? If we stick with this default configuration, we will miss unused private variables (prefixed with underscore), and thus miss any potential issue.
> 
> I believe they only ignored them for F841. We could fix this by overriding [dummy-variable-rgx](https://docs.astral.sh/ruff/settings/#lint_dummy-variable-rgx)

Setting:
```
[tool.ruff.lint]
# Only ignore variables named "_".
dummy-variable-rgx = "^_$"
```

I get only one more issue:
`source\virtualBuffers\lotusNotes.py:112:6: F811 Redefinition of unused `_searchableAttribsForNodeType` from line 54`

Though, I do not know this file at all, nor how to test it. And I am not able to `git blame` the changes in this files: I only get the Ruff and line ending reformatting commits (b31c94cf58 and 8fb8ffcaba).

@seanbudd, without any other hint, I am tempted to leave this file as is, to keep this potential history, due to impossibility of testing.

Or maybe should I use `git blame` with a specific parameter to ignore these commits? I thought that it was automatically configured in the repo... Could you help me with this?

Also @seanbudd, do you think that I should modify `ruff/settings/#lint_dummy-variable-rgx` permanently, i.e. in this PR?

I'll switch this PR to Ready to get the required answered.

File: source/NVDAObjects/window/edit.py [hunk 3]
```diff
@@ -421,12 +421,12 @@ def _getSelectionOffsets(self):
 		else:
 			start = ctypes.c_uint()
 			end = ctypes.c_uint()
-			res = watchdog.cancellableSendMessage(  # noqa: F841
+			watchdog.cancellableSendMessage(
```
Comment (seanbudd) [inline source/NVDAObjects/window/edit.py:424]: should we be checking the result here and logging a failure?

File: source/NVDAObjects/window/edit.py [hunk 4]
```diff
@@ -840,7 +840,7 @@ def _getEmbeddedObjectLabel(self, embedRangeObj):
 			dataObj = None
 		if dataObj:
 			text = comtypes.BSTR()
-			res = NVDAHelper.localLib.getOleClipboardText(dataObj, ctypes.byref(text))  # noqa: F841
+			NVDAHelper.localLib.getOleClipboardText(dataObj, ctypes.byref(text))
```
Comment (seanbudd) [inline source/NVDAObjects/window/edit.py:843]: should we be checking the result here and logging a failure?

File: source/NVDAObjects/window/winword.py [hunk 5]
```diff
@@ -966,8 +966,8 @@ def _normalizeControlField(self, field):
 		return newField
 
 	def _normalizeFormatField(self, field, extraDetail=False):
-		_startOffset = int(field.pop("_startOffset"))
-		_endOffset = int(field.pop("_endOffset"))
+		field.pop("_startOffset")
+		field.pop("_endOffset")
```
Comment (seanbudd) [inline source/NVDAObjects/window/winword.py:970]: should these just be removed entirely?
These seem unused in NVDA.

This line should probably be removed too

https://github.com/nvaccess/nvda/blob/8fb8ffcaba3950b1e4ca7b58e165267fc5369eb9/nvdaHelper/remote/winword.cpp#L1293

Thoughts @michaelDCurran ?

**Comment by seanbudd** [comment]
>I get only one more issue:
source\virtualBuffers\lotusNotes.py:112:6: F811 Redefinition of unused _searchableAttribsForNodeType from line 54

I think it would be best to leave this. It might be worth adding more thorough comments noting that the second definition is the one used, however we may want to migrate functionality from the first definition.

>Or maybe should I use git blame with a specific parameter to ignore these commits? I thought that it was automatically configured in the repo... Could you help me with this?

You may need to add .git-blame-ignore-revs in your local git config, or use the CLI flag

```
git blame --ignore-revs-file .git-blame-ignore-revs
git config blame.ignoreRevsFile .git-blame-ignore-revs
```

>do you think that I should modify ruff/settings/#lint_dummy-variable-rgx permanently, i.e. in this PR?

I think there are probably cases where we want to follow the magic rules for a dummy variable.
Perhaps this should be considered in a separate PR. 
We could make variable names triple-prefixed with underscores dummy variables for example.

File: source/NVDAObjects/window/edit.py (hunk 4 above)
Comment (CyrilleB79) [reply source/NVDAObjects/window/edit.py:843]: I do not know how frequently this call can fail.
If I log an error, is there a risk that I log many errors here?

Also, looking at function `cancellableSendMessageTimeout` (in file `nvdaHelperLocal.cpp`), it's not clear to me if I should only look at return code 0 and look at last error, or if some other non-zero return codes also need to be logged in the Python side of the code.

File: source/NVDAObjects/window/winword.py (hunk 5 above)
Comment (michaelDCurran) [reply source/NVDAObjects/window/winword.py:970]: Yes, those attributes can be removed from the xml stream. They used to be used to recreate an MS Word range object to fetch extra track change revision info such as author and date in _normalizeFormatField, but we no longer report that info here.

Comment (michaelDCurran) [reply source/NVDAObjects/window/winword.py:970]: The code that used them was removed in ca50d9b004 sometime after 2015.4.

Comment (seanbudd) [reply source/NVDAObjects/window/winword.py:970]:
```suggestion
```

Can you also remove the line in nvdaHelper?

**Comment by seanbudd** [comment]
@CyrilleB79 - do you plan to continue this?

File: source/NVDAObjects/window/winword.py (hunk 5 above)
Comment (CyrilleB79) [reply source/NVDAObjects/window/winword.py]: > Can you also remove the line in nvdaHelper?

Actually, I have not removed all the line but rather only `_startOffset` and `_endOffset` value in the xml stream.

**Comment by CyrilleB79** [comment]
@seanbudd, finally, it is ready again.

**Review by seanbudd — APPROVED** [review]

**Pull request merged**